tests/onegov/form/test_collection.py

```diff
@@ -101,6 +101,35 @@ def test_submission_extra_data(session):
     assert submission.email == 'bill.lumbergh@initech.com'
 
 
+def test_submission_update(session):
+    collection = FormCollection(session)
+
+    form = collection.definitions.add('Whatever', definition=dedent("""
+        First Name * = ___
+        Last Name * = ___
+        E-Mail = @@@
+        Date = YYYY.MM.DD
+    """))
+
+    data = MultiDict([
+        ('first_name', 'Bill'),
+        ('last_name', 'Gate.'),
+        ('e_mail', 'bill.gates@microsoft.com'),
+        ('date', '1955-10-28')
+    ])
+
+    submitted_form = form.form_class(data)
+    submission = collection.submissions.add(
+        'whatever', submitted_form, state='complete')
+    assert submission.title == 'Bill, Gate.'
+
+    edited_form = submitted_form
+    edited_form.first_name.data = '<Billy>Bill'
+    edited_form.last_name.data = 'Gates'
+    collection.submissions.update(submission, edited_form)
+    assert submission.title == 'Bill, Gates'
```

Comment:
Technically `<Billy>` being stripped from the title is a bug, since it will be rendered as `&lt;Billy&gt;` and only unescape makes it look like a tag, but it would be fairly difficult to avoid it. So it's probably not worth fixing.

I'm frankly not quite sure why we unescape here anyways, unless there are some fields that are allowed to contain Markup.

Reply:
It's possible that we render the title with `structure` for whatever reason, so you can leave it like it for now but I'd make a `FIXME` comment to remind me to investigate this when I look at consistent use of `Markup`.

Reply:
Agree, I played/tried quite a bit but also do not fully understand why its done this way.
I will revert the last commit and leave this alone...